src/js/components/AppBody.vue

```diff
@@ -25,22 +25,146 @@
+    'main-menu': MainMenu,
+    'split-pane': SplitPane,
+    'single-image': SingleImage,
+    'editor': Editor,
+    'combination': Combination,
+  },
+  data() {
+    return {
+      keepMenuOpen: false,
+      mouseOver: false,
+      corpus: Corpus,
+      menuLoaded: false,
+    }
+  },
+  computed: {
+    menuOpen() {
+      return this.mouseOver || this.keepMenuOpen
+    },
+  },
+  created() {
+    // Create and populate the corpus model, with existing data
+    // from the router.
+    this.$store.commit('addWorking')
+    this.corpus = new Corpus(this.$store.state.sessionID, this.$store.state.userID)
+    this.corpus.populateCombinations()
+    .then(res => {
+      this.menuLoaded = true
+      this.$store.commit('delWorking')
+      if (this.$route.params.scrollID && this.$route.params.scrollID !== '~') {
+        this.$store.commit('addWorking')
+        this.corpus.populateColumnsOfCombination(
+          this.$route.params.scrollID,
+          this.$route.params.scrollVersionID
+        )
+        this.corpus.populateImageReferencesOfCombination(this.$route.params.scrollVersionID)
+        .then(res1 => {
+          this.$store.commit('delWorking')
+          if(this.$route.params.imageID !== '~') {
+            this.$store.commit('addWorking')
+            this.corpus.populateImagesOfImageReference(
+              this.$route.params.imageID, 
+              this.$route.params.scrollVersionID
+            )
+            .then(res2 => {
+              this.$store.commit('delWorking')
+              if(this.$route.params.artID !== '~') {
+                this.$store.commit('addWorking')
+                this.corpus.populateArtefactsOfImageReference(
+                  this.$route.params.imageID, 
+                  this.$route.params.scrollVersionID
+                )
+                .then(res3 => {
+                  this.$store.commit('delWorking')
+                  this.resetRouter()
+                })
+              } else {
+                this.resetRouter()
+              }
+            })
+          } else {
+            this.resetRouter()
+          }
+        })
+      }
+    })
+  },
+  methods: {
+    resetRouter() {
+      // Trigger a router change here, so we don't need extra mount()
+      // functions in all of our vue components.  Is there a more 
+      // elegant way to achieve this?
+      const scrollID = this.$route.params.scrollID
+      const scrollVersionID = this.$route.params.scrollVersionID
+      const colID = this.$route.params.colID
+      const imageID = this.$route.params.imageID
+      const artID = this.$route.params.artID
+
+      this.$router.push({name: 'workbenchAddress',
+        params: {
+          scrollID: '~',
+          scrollVersionID: '~',
+          colID: '~',
+          imageID: '~',
+          artID: '~'
+        }
+      }, () => {
+        // I don't know why the listeners don't pick up on the 
+        // router change unless I give a tiny delay.
```

Comment:
maybe try taking a look at Vue.nextTick: https://vuejs.org/v2/api/#Vue-nextTick. Dunno if that will help but it might.

Why do we need to do two router pushes back to back here?

Reply:
This is perhaps a bit of a hack, but it can minimize code in the individual componenents, since they no longer need a created()/mount() method in addition to our watch: method for the router.  On startup of the website, by flipping the router from the original route, to the blanks with ~, and then to the original address, the components' watchers pick up on the change and execute any code in their router watcher.  I think this reduces possibilities for errors, when you change something in the router watcher and forget to make an update to the created()/mount() method.  What do you think of this scheme?
I will take a look at nextTick, that would indeed be better than using a timeout.